source/characterProcessing.py

```diff
@@ -779,9 +756,156 @@ def clearSpeechSymbols():
 def handlePostConfigProfileSwitch(prevConf=None):
 	if not prevConf:
 		return
-	if prevConf["speech"]["includeCLDR"] is not config.conf["speech"]["includeCLDR"]:
-		# Either included or excluded CLDR data, so clear the cache.
+	if set(prevConf["speech"]["symbolDictionaries"]) != set(config.conf["speech"]["symbolDictionaries"]):
+		# Either included or excluded dictionaries, so clear the cache.
 		clearSpeechSymbols()
 
 
-config.post_configProfileSwitch.register(handlePostConfigProfileSwitch)
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class SymbolDictionaryDefinition:
+	name: str
+	"""The name of the dictionary."""
+	path: str
+	"""The path to the dictionary.
+	This should be a formattable string, where {locale} is replaced by the locale to fetch a dictionary for.
+	"""
+	displayName: str | None = None
+	"""The translatable name of the dictionary.
+	When not provided, the dictionary can not be visible to the end user.
+	"""
+	allowComplexSymbols: bool = False
+	"""Whether this dictionary allows complex symbols."""
+	mandatory: bool = False
+	"""Whether this dictionary is mandatory.
+	Mandatory dictionaries are always enabled."""
+	user: bool = False
+	"""Whether this is a user dictionary definition."""
+	symbols: LocaleDataMap[SpeechSymbols] = dataclasses.field(init=False, repr=False, compare=False)
+
+	def __post_init__(self):
+		if self.path.count("{locale}") != 1:
+			raise ValueError(f"Invalid formatable path for dictionary: {self.path!r}")
+		if not self.displayName and not self.mandatory:
+			raise ValueError("A nonmandatory dictionary without a display name is unsupported")
+		object.__setattr__(self, "symbols", LocaleDataMap(self._getSymbols))
+
+	@cached_property
+	def userVisible(self) -> bool:
+		"""Whether this dictionary is visible to end users (i.e. in the GUI).
+		Mandatory dictionaries are hidden.
+		"""
+		return not self.mandatory and bool(self.displayName)
+
+	@property
+	def enabled(self) -> bool:
+		return self.mandatory or self.name in config.conf["speech"]["symbolDictionaries"]
+
+	def getSymbols(self, locale: str) -> SpeechSymbols:
+		"""Gets the symbols for a given locale.
+		:param locale: The locale to get symbols for.
+		:raises FileNotFoundError: When this is not a user dictionary and the locale wasn't found.
+		"""
+		return self.symbols.fetchLocaleData(locale, fallback=False)
+
+	def _getSymbols(self, locale: str) -> SpeechSymbols:
```

Comment:
should this be init symbols instead? it seems this is called once on initialization
```suggestion
	def _initSymbols(self, locale: str) -> SpeechSymbols:
```